### PR TITLE
Refactor `plotRaw` function in ISISEnergyTransferModel

### LINKED ISSUE #35963: Refactor plotRaw in ISIS Energy Transfer Model
**Describe the outcome that is desired.**

The ISIS Energy Transfer tab has been refactored into MVP (see https://github.com/mantidproject/mantid/issues/35912). `plotRaw` In the `IETModel.h` class (located in ISISEnergyTransferView) is a big function that runs several algorithms. This makes testing the function hard. The goal of this issue is to refactor the function into multiple smaller testable functions and add unit tests to them.
**Describe any solutions you are considering**

Refactor every algorithm step into a separate function then build the algorithm pipeline in the main function using the bath algorithm runner.

### PR DESCRIPTION
**Pull request by robertapplin**
### Description of work
This PR refactors the `plotRaw` function in ISISEnergyTransferModel to make it smaller and easier to maintain. I have made a few changes with regards to this:
- Created a `ReductionAlgorithmUtils` file. This file contains function which create and configure an algorithm and its properties ready to be executed by the BatchAlgorithmRunner
- The refactored function in ISISEnergyTransferModel now returns a queue of algorithms, which then gets executed by the BatchAlgorithmRunner held by the presenter
- A large number of tests were added
- A few utility functions were added to the `ISISEnergyTransferModelUtils` file.

Fixes #35963 

### To test:

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: qt/widgets/common/inc/MantidQtWidgets/Common/ConfiguredAlgorithm.h [hunk 1]
```diff
@@ -14,16 +14,19 @@ namespace MantidQt::API {
 class EXPORT_OPT_MANTIDQT_COMMON ConfiguredAlgorithm : public IConfiguredAlgorithm {
 public:
   ConfiguredAlgorithm(Mantid::API::IAlgorithm_sptr algorithm,
-                      std::unique_ptr<Mantid::API::IAlgorithmRuntimeProps> properties);
+                      std::unique_ptr<Mantid::API::IAlgorithmRuntimeProps> properties,
```
Comment (MohamedAlmaki) [inline qt/widgets/common/inc/MantidQtWidgets/Common/ConfiguredAlgorithm.h:17]: Why this change has been introduced?

**Review by MohamedAlmaki — COMMENTED** [review]
Looks good and clean. Just a few enquires

File: qt/widgets/common/inc/MantidQtWidgets/Common/ConfiguredAlgorithm.h (hunk 1 above)
Comment (robertapplin) [reply qt/widgets/common/inc/MantidQtWidgets/Common/ConfiguredAlgorithm.h:17]: This has been introduced to allow us to queue up the `Load` algorithm in the BatchAlgorithmRunner. This algorithm is slightly different to many others in that it will determine the appropriate load algorithm to use on runtime. This means we cannot validate the properties prior to the execution of the algorithm because if we do then it will only check if the properties exist on the `Load` algorithm, and not if they exist on the specialisation of the load algorithm it will actually choose. So adding this option allows us to skip the extra validation step in the BatchAlgorithmRunner at line 251

File: qt/scientific_interfaces/Indirect/Reduction/ISISEnergyTransferModel.cpp [hunk 2]
```diff
@@ -216,73 +216,35 @@ std::vector<std::string> IETModel::validatePlotData(IETPlotData const &plotParam
...
-  inputFromRebin->setPropertyValue("InputWorkspace", name);
+  return plotRawAlgorithmQueue(rawFile, basename, instData.getInstrument(), detectorList,
+                               plotParams.getBackgroundData());
+}
 
-  std::vector<specnum_t> detectorList;
-  for (specnum_t i = spectraMin; i <= spectraMax; i++)
-    detectorList.emplace_back(i);
+std::deque<MantidQt::API::IConfiguredAlgorithm_sptr>
+IETModel::plotRawAlgorithmQueue(std::string const &rawFile, std::string const &basename,
```
[the remark this answers is not in the document]
Comment (robertapplin) [reply qt/scientific_interfaces/Indirect/Reduction/ISISEnergyTransferModel.cpp:231]: When working on the ALFView interface, I came across the same problem - whether it was best to run the BatchAlgorithmRunner from the presenter or from the model. In the end I decided it was better if the presenter owned and called this runner because it is a Qt object, and it felt more natural that the presenter be used as the communication piece between the model and anything else that is Qt related such as the View or BatchAlgorithmRunner. As you say, it also provides a better separation of concerns.

The `ALFInstrumentPresenter` and `ALFAnalysisPresenter` provide an example of something similar to the pattern I was trying to copy a little bit. (The only difference is that they own an `ALFALgorithmManager`, and then this manager owns the BatchAlgorithmRunner)

[the remark this answers is not in the document]
Comment (MohamedAlmaki) [reply qt/scientific_interfaces/Indirect/Reduction/ISISEnergyTransferModel.cpp:231]: Oh, I've noticed it is a Qt object. Yes, that's make sense

File: qt/scientific_interfaces/Indirect/test/ISISEnergyTransferModelTest.h [hunk 3]
```diff
@@ -613,6 +613,30 @@ class ISISEnergyTransferModelTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS(errors.size(), 0);
   }
 
+  void test_plotRawAlgorithmQueue_returns_two_algorithms_if_remove_background_is_false() {
```
Comment (MohamedAlmaki) [inline qt/scientific_interfaces/Indirect/test/ISISEnergyTransferModelTest.h:616]: here the naming is also inconsistent

**Review by MohamedAlmaki — APPROVED** [review]
Thanks for the changes and the clean refactor, happy to approve
